src/Microsoft.Azure.WebJobs.Host/Bindings/QueueBindingRules.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Azure.WebJobs.Host.Queues;
+using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Storage.Queue;
+using Microsoft.WindowsAzure.Storage.Queue;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    // Write up bindinging rules for [Queue] attribute. 
+    // This is fundemantentally an IAsyncCollector<IStorageQueueMessage>
+    internal class QueueBindingRules
```

Comment:
This will be an example we point people to to follow, so I'd like to make sure we have good naming that we recommend people to follow.

What if we called this QueueBindingProvider, and it had a method Build? That makes sense to me. Maybe we steer people to a factory pattern, with a static QueueBindingProvider.Build?

Having a "build rules" method that returns an IBindingProvider seems off to me.